engine/core/Engine.cpp

```diff
@@ -230,3 +236,28 @@ bool Engine::isStarted() {
 double Engine::getDeltaTime() {
     return Engine::currentTime - Engine::lastTime;
 }
+
+std::string Engine::getConfigDir() {
+    // get the home directory and load the cache from there
+    // Here is my most convoluted ifdef set ever
+    std::string pathString;
+#ifndef _WIN32
+    const char *homePath = getenv("HOME");
+    if (!homePath) {
+        struct passwd* pwd = getpwuid(getuid());
+        if (pwd)
+            homePath = pwd->pw_dir;
+    }
+    pathString.append(homePath);
+#if defined(__APPLE__) && defined(__MACH__) // Assume macOS
+    pathString.append("/Library/Application Support/ChiraEngine/");
+#else // Assume Linux
+    pathString.append("/.local/ChiraEngine");
+#endif // end this linux macos differences def
+#else // We must be on windows
+    // Get the environment vairable for the user on windows
+    pathString = getenv("USERPROFILE");
+    pathString.append("\\Application Data\\Roaming\\ChiraEngine\\");
+#endif
```

Comment:
About the structure of ifs here and a few other places:

Rather than having `#else`s, I personally like testing explicitly for all platforms. Then, you could have an `#else` clause with an `#error Unsupported platform!` directive. If the engine is ever ported to some weird OS that would save lots of time

Reply:
will do. this was written on a vnc session with shitty connection so I cut some corners on the implementation!